mavros/src/lib/uas_sensor_orientation.cpp

```diff
@@ -26,66 +25,87 @@ typedef std::pair<const std::string, const Eigen::Quaterniond> OrientationPair;
+/* 10 */ make_orientation("ROLL_180_YAW_90", 			180.0,   0.0,  90.0),
+/* 11 */ make_orientation("ROLL_180_YAW_135", 			180.0,   0.0, 135.0),
+/* 12 */ make_orientation("PITCH_180", 				  0.0, 180.0,   0.0),
+/* 13 */ make_orientation("ROLL_180_YAW_225", 			180.0,   0.0, 225.0),
+/* 14 */ make_orientation("ROLL_180_YAW_270", 			180.0,   0.0, 270.0),
+/* 15 */ make_orientation("ROLL_180_YAW_315", 			180.0,   0.0, 315.0),
+/* 16 */ make_orientation("ROLL_90", 			 	 90.0,   0.0,   0.0),
+/* 17 */ make_orientation("ROLL_90_YAW_45", 			 90.0,   0.0,  45.0),
+/* 18 */ make_orientation("ROLL_90_YAW_90", 			 90.0,   0.0,  90.0),
+/* 19 */ make_orientation("ROLL_90_YAW_135", 			 90.0,   0.0, 135.0),
+/* 20 */ make_orientation("ROLL_270", 				270.0,   0.0,   0.0),
+/* 21 */ make_orientation("ROLL_270_YAW_45", 			270.0,   0.0,  45.0),
+/* 22 */ make_orientation("ROLL_270_YAW_90", 			270.0,   0.0,  90.0),
+/* 23 */ make_orientation("ROLL_270_YAW_135", 			270.0,   0.0, 135.0),
+/* 24 */ make_orientation("PITCH_90", 				  0.0,  90.0,   0.0),
+/* 25 */ make_orientation("PITCH_270", 				  0.0, 270.0,   0.0),
+/* 26 */ make_orientation("PITCH_180_YAW_90", 			  0.0, 180.0,  90.0),
+/* 27 */ make_orientation("PITCH_180_YAW_270", 			  0.0, 180.0, 270.0),
+/* 28 */ make_orientation("ROLL_90_PITCH_90", 			 90.0,  90.0,   0.0),
+/* 29 */ make_orientation("ROLL_180_PITCH_90", 			180.0,  90.0,   0.0),
+/* 30 */ make_orientation("ROLL_270_PITCH_90", 			270.0,  90.0,   0.0),
+/* 31 */ make_orientation("ROLL_90_PITCH_180", 			 90.0, 180.0,   0.0),
+/* 32 */ make_orientation("ROLL_270_PITCH_180", 		270.0, 180.0,   0.0),
+/* 33 */ make_orientation("ROLL_90_PITCH_270", 			 90.0, 270.0,   0.0),
+/* 34 */ make_orientation("ROLL_180_PITCH_270", 		180.0, 270.0,   0.0),
+/* 35 */ make_orientation("ROLL_270_PITCH_270", 		270.0, 270.0,   0.0),
+/* 36 */ make_orientation("ROLL_90_PITCH_180_YAW_90", 		 90.0, 180.0,  90.0),
+/* 37 */ make_orientation("ROLL_90_YAW_270", 			 90.0,   0.0, 270.0),
+/* 38 */ make_orientation("ROLL_315_PITCH_315_YAW_315",		315.0, 315.0, 315.0)
 }};
 
+std::string UAS::str_sensor_orientation(MAV_SENSOR_ORIENTATION orientation)
+{
+	size_t idx = size_t(orientation);
+	if (idx >= sensor_orientations.size()) {
+		ROS_WARN_NAMED("uas", "SENSOR: wrong orientation index: %zu", idx);
+		return std::to_string(idx);
+	}
+
+	return sensor_orientations[idx].first;
+}
+
 Eigen::Quaterniond UAS::sensor_orientation_matching(MAV_SENSOR_ORIENTATION orientation)
 {
 	size_t idx = size_t(orientation);
 	if (idx >= sensor_orientations.size()) {
-		ROS_WARN_NAMED("uas", "SENSOR: wrong orintation index: %zu", idx);
+		ROS_WARN_NAMED("uas", "SENSOR: wrong orientation index: %zu", idx);
 		return Eigen::Quaterniond::Identity();
 	}
 
 	return sensor_orientations[idx].second;
 }
+
+int UAS::orientation_from_str(std::string sensor_orientation)
+{
+	int index, idx = 0;
+	for (const auto &orientation : sensor_orientations) {
```

Comment:
Here you may use regular for, not foreach.
Also not this function do not work, almost always return -1 because it should look that (pseudocode):
```cpp
for (size_t idx = 0; idx < so.size(); idx++) {
 if (so[idx].first == key)
  return idx;
}
// fallback
v = strtol(key); // this should not fail, so use try {} catch

// not found at all
return -1;
```